### PR TITLE
Add support for SHAKE key derivation

### PR DESCRIPTION
**Pull request by ifranzki**

### CODE REVIEW

File: testcases/crypto/digest_func.c [hunk 1]
```diff
@@ -1694,6 +1696,16 @@ CK_RV do_SHA_derive_key(void)
                 derive_mechs[i].mechanism == CKM_SHA1_KEY_DERIVATION)
                 continue;
 
+            /*
+             * SHAKE dosn't have a default digest size. so key type and/or
```
Comment (jschmidb) [inline testcases/crypto/digest_func.c:1700]: doesn't

File: usr/lib/ica_s390_stdll/ica_specific.c [hunk 2]
```diff
@@ -1818,6 +1850,155 @@ CK_RV token_specific_sha_final(STDLL_TokData_t *tokdata, DIGEST_CONTEXT *ctx,
...
+        TRACE_ERROR("Failed to build the attribute from CKA_VALUE, rc=%s.\n",
+                    ock_err(rc));
+        goto out;
+    }
+
+    switch (derived_key_type) {
+    case CKK_GENERIC_SECRET:
+    case CKK_AES:
+    case CKK_AES_XTS:
+        /* Supply CKA_VAUE_LEN since this is required for those key types */
```
[the remark this answers is not in the document]
Comment (ifranzki) [reply usr/lib/ica_s390_stdll/ica_specific.c]: fixed

File: testcases/crypto/digest_func.c (hunk 1 above)
Comment (ifranzki) [reply testcases/crypto/digest_func.c]: fixed

**Review by jschmidb — APPROVED** [review]
lgtm

**Pull request merged**